File: box-admin-web/src/api/agentTemplate.ts
import http, { type Result } from './http'

export interface AgentTemplateVO {
  id: number
  templateCode: string
  name: string
  description?: string
  avatarUrl?: string
  category?: string
  systemPrompt?: string
  platformModelId?: number
  platformModelName?: string
  temperature?: number
  topP?: number
  maxTokens?: number
  streamEnabled?: boolean
  status: string
  sortOrder?: number
  installCount?: number
  createdAt?: string
  updatedAt?: string
}

export function fetchAgentTemplates() {
  return http.get<Result<AgentTemplateVO[]>>('/agent-templates')
}

export function createAgentTemplate(payload: {
  templateCode: string
  name: string
  description?: string
  category?: string
  systemPrompt?: string
  platformModelId: number
  temperature?: number
  topP?: number
  maxTokens?: number
  streamEnabled?: boolean
  sortOrder?: number
}) {
  return http.post<Result<AgentTemplateVO>>('/agent-templates', payload)
}

export function updateAgentTemplate(
  id: number,
  payload: {
    name: string
    description?: string
    category?: string
    systemPrompt?: string
    platformModelId: number
    temperature?: number
    topP?: number
    maxTokens?: number
    streamEnabled?: boolean
    sortOrder?: number
  },
) {
  return http.put<Result<AgentTemplateVO>>(`/agent-templates/${id}`, payload)
}

export function updateAgentTemplateStatus(id: number, status: string) {
  return http.put<Result<AgentTemplateVO>>(`/agent-templates/${id}/status`, { status })
}

File: box-server/box-domain/src/main/java/com/boxai/domain/agent/AgentTemplate.java
package com.boxai.domain.agent;

import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Setter
public class AgentTemplate {

    private Long id;
    private String templateCode;
    private String name;
    private String description;
    private String avatarUrl;
    private String category;
    private String systemPrompt;
    private Long platformModelId;
    private BigDecimal temperature;
    private BigDecimal topP;
    private Integer maxTokens;
    private Boolean streamEnabled;
    private String status;
    private String reviewStatus;
    private String visibility;
    private String tenantIdsJson;
    private Integer rolloutPercent;
    private Integer sortOrder;
    private Integer installCount;
    private Long createdBy;
    private Long updatedBy;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
